### PR TITLE
Adls storage added to sandbox

### PR DESCRIPTION
**Pull request by kmcclenn**
## Summary

Adds ADLS integration with the sandbox. Tables created with the tables service now stored in an ADLS Container.

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [ ] Bug Fixes
- [x] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [ ] Refactoring
- [ ] Documentation
- [ ] Tests

Tables are stored using the storage account and blob container provisioned in the sandbox.

The PR also involves copying the Iceberg Azure files due to conflict issues. Once Openhouse is upgraded to Iceberg >= 1.4.0, we can remove the copied files. More package details found in the description of PR #120 

## Testing Done

- [x] Manually Tested on local setup. Please include commands ran, and their output.
- [ ] Added new tests for the changes made.
- [ ] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

Ran the following commands from the root directory:
```
./gradlew clean build -x test
cd infra/recipes/terraform/azure/environments/container
terraform apply
cd ../sandbox
terraform apply
```
to build the images and provision the sandbox, and then upon completion
```
az aks get-credentials --resource-group <resource-group-name> --name <k8s-cluster-name>
kubectl get pods
kubectl port-forward <tables-pod-name> 8080:8080
```
to import the pods and set up port forwarding. Then, made various http requests to the tables service and verified that the created tables were showing up in the Azure storage container. See the video below for an example.

https://github.com/user-attachments/assets/71aed26d-6438-406a-8064-87e37efe06d4

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

File: iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOConfig.java [hunk 1]
```diff
@@ -66,29 +66,21 @@ FileIO provideLocalFileIO() {
...
-        // Spring doesn't define the bean if the return value is null
-        log.debug("S3 storage type is not configured", err);
-        return null;
-      }
-      // S3 storage configured but could not instantiate S3FileIO.
-      throw new IllegalArgumentException(
-          "S3 storage configured but could not initialize S3FileIO" + e.getMessage());
+      // If the ADLS storage type is not configured, return null
+      // Spring doesn't define the bean if the return value is null
+      return null;
```
Comment (sumedhsakdeo) [inline iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOConfig.java:83]: Why was provideS3FileIO removed?

**Review by sumedhsakdeo — CHANGES_REQUESTED** [review]
Revert the removals of S3 support.

**Comment by HotSushi** [comment]
Is this PR still relevant? similar code changes were checkedin as part of #120

File: iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOConfig.java (hunk 1 above)
Comment (kmcclenn) [reply iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOConfig.java:83]: Ill fix

Comment (kmcclenn) [reply iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/fileio/FileIOConfig.java]: Fixed

File: infra/recipes/k8s/helm/tables/config/azure/sandbox/cluster.yaml [hunk 2]
```diff
@@ -2,14 +2,23 @@ cluster:
...
+    root-path: "/data/openhouse"
+  storages:
+    default-type: "adls"
+    types:
+      adls:
+        rootpath: <container-name>@<storage-account-name>.dfs.core.windows.net 
+        endpoint: "abfs://"
+        parameters:
+          adls.auth.shared-key.account.name: <storage-account-name>
+          adls.auth.shared-key.account.key: <storage-account-key>
```
Comment (sumedhsakdeo) [inline infra/recipes/k8s/helm/tables/config/azure/sandbox/cluster.yaml:15]: Nit: Is this updated in README.

**Review by sumedhsakdeo — APPROVED** [review]

**Review by abhisheknath2011 — APPROVED** [review]

File: infra/recipes/k8s/helm/tables/config/azure/sandbox/cluster.yaml (hunk 2 above)
Comment (kmcclenn) [reply infra/recipes/k8s/helm/tables/config/azure/sandbox/cluster.yaml:15]: Ill make sure it is

Comment (kmcclenn) [reply infra/recipes/k8s/helm/tables/config/azure/sandbox/cluster.yaml:15]: Actually nevermind no action is needed from the user, it is all replaced automatically by the helm templates

**Pull request merged**